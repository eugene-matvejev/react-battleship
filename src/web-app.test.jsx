import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _WebApp from './web-app';

const { WrappedComponent: WebApp } = _WebApp;

configure({ adapter: new Adapter() });

describe('<WebApp/>', () => {
    const props = {
        title: '{{title}}',
        authorizedRoutes: [
            {
                c: () => <span className="authorized" />,
                path: '/',
                props: {
                },
            },
        ],
        unauthorizedRoutes: [
            {
                c: () => <span className="unauthorized" />,
                path: '/',
                props: {
                },
            },
        ],
    };
    const user = { email: 'example@example.com' };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<WebApp {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['user', user],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<WebApp {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });

    describe('internal callbacks', () => {
        describe('::onAuthenticate', () => {
            it('should set state field [::user] from payload', () => {
                const spy = spyOn(WebApp.prototype, 'setState');

                shallow(<WebApp {...props} />).instance().onAuthenticate(user);

                expect(spy).toBeCalledWith({ user });
            });
        });
    });
});
