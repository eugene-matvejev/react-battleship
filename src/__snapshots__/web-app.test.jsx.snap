// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebApp/> render with default/required props 1`] = `
<Fragment>
  <SideNav
    className=""
    data-cy=""
    routes={Array []}
    title="{{title}}"
  />
  <main
    className="webapp_content"
  >
    <TopNav
      className=""
      data-cy=""
      initials="CY"
    />
    <Switch>
      <Route
        component={[Function]}
        key="0"
        path="/"
      />
    </Switch>
  </main>
</Fragment>
`;

exports[`<WebApp/> render with optional props [::user] as "[object Object]" 1`] = `
<Fragment>
  <SideNav
    className=""
    data-cy=""
    routes={Array []}
    title="{{title}}"
  />
  <main
    className="webapp_content"
  >
    <TopNav
      className=""
      data-cy=""
      initials="CY"
    />
    <Switch>
      <Route
        component={[Function]}
        key="0"
        path="/"
      />
    </Switch>
  </main>
</Fragment>
`;
